drivers/mtd_flashpage/mtd_flashpage.c

```diff
@@ -26,87 +28,128 @@
 #include "architecture.h"
 #include "cpu.h"
 #include "cpu_conf.h"
+#include "macros/utils.h"
 #include "mtd_flashpage.h"
 #include "periph/flashpage.h"
 
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
 #define MTD_FLASHPAGE_END_ADDR     ((uint32_t) CPU_FLASH_BASE + (FLASHPAGE_NUMOF * FLASHPAGE_SIZE))
 
 static int _init(mtd_dev_t *dev)
 {
-    (void)dev;
+    mtd_flashpage_t *super = container_of(dev, mtd_flashpage_t, base);
+    (void)super;
     assert(dev->pages_per_sector * dev->page_size == FLASHPAGE_SIZE);
+    assert(!(super->offset % dev->pages_per_sector));
+    assert(super->offset + dev->pages_per_sector * dev->page_size * dev->sector_count
```

Comment:
Sorry, I´m still confused here.
```C
assert(super->offset /* this is a page */
       + dev->pages_per_sector * dev->page_size * dev->sector_count /* this is maximum size in bytes relative to super->offset */
       <= MTD_FLASHPAGE_END_ADDR /* this is a raw address in flash */);
```

If we compare with `MTD_FLASHPAGE_END_ADDR`, must not `CPU_FLASH_BASE` play a role somehow?
Must it be `flashpage_addr(super->offset / dev->pages_per_sector)`?

Reply:
You are of course right - thanks again!